src/object-store/src/layers/lru_cache/read_cache.rs

```diff
@@ -135,24 +136,22 @@ impl<C: Accessor + Clone> ReadCache<C> {
     pub(crate) async fn recover_cache(&self) -> Result<(u64, u64)> {
         let (_, mut pager) = self.file_cache.list("/", OpList::default()).await?;
 
-        while let Some(entries) = pager.next().await? {
-            for entry in entries {
-                let read_key = entry.path();
+        while let Some(entry) = pager.next().await? {
+            let read_key = entry.path();
 
-                // We can't retrieve the metadata from `[opendal::raw::oio::Entry]` directly,
-                // because it's private field.
-                let size = {
-                    let stat = self.file_cache.stat(read_key, OpStat::default()).await?;
+            // We can't retrieve the metadata from `[opendal::raw::oio::Entry]` directly,
+            // because it's private field.
+            let size = {
+                let stat = self.file_cache.stat(read_key, OpStat::default()).await?;
 
-                    stat.into_metadata().content_length()
-                };
+                stat.into_metadata().content_length()
+            };
 
-                OBJECT_STORE_LRU_CACHE_ENTRIES.inc();
-                OBJECT_STORE_LRU_CACHE_BYTES.add(size as i64);
-                self.mem_cache
-                    .insert(read_key.to_string(), ReadResult::Success(size as u32))
-                    .await;
-            }
+            OBJECT_STORE_LRU_CACHE_ENTRIES.inc();
+            OBJECT_STORE_LRU_CACHE_BYTES.add(size as i64);
+            self.mem_cache
+                .insert(read_key.to_string(), ReadResult::Success(size as u32))
+                .await;
```

Comment:
Consider refactoring it with [list_with_metakey](https://github.com/apache/incubator-opendal/blob/035b3dc48a26ac1350c0ba12872f9d97d602840c/core/src/docs/rfcs/2779_list_with_metakey.md) and [concurrent_stat_in_list](https://github.com/apache/incubator-opendal/blob/035b3dc48a26ac1350c0ba12872f9d97d602840c/core/src/docs/rfcs/3574_concurrent_stat_in_list.md)